Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -437,6 +436,23 @@ public class CarPlayNavigationViewController: UIViewController {
         let alert = CPActionSheetTemplate(title: arrivalTitle, message: arrivalMessage, actions: [rateAction, exitAction])
         carInterfaceController.presentTemplate(alert, animated: true)
     }
+    
+    func presentWaypointArrivalUI(for waypoint: Waypoint) {
+        var title = NSLocalizedString("CARPLAY_ARRIVED", bundle: .mapboxNavigation, value: "You have arrived", comment: "Title on arrival action sheet")
+        if let name = waypoint.name {
+            title = name
+        }
+        
+        let continueTitle = NSLocalizedString("CARPLAY_CONTINUE", bundle: .mapboxNavigation, value: "Continue", comment: "Title on continue button in CarPlay")
+        let continueAlert = CPAlertAction(title: continueTitle, style: .default) { (action) in
+            self.navigationService.routeProgress.legIndex += 1
```

Comment:
I think this is one of the possible scenarios, which might cause crash in https://github.com/mapbox/mapbox-navigation-ios/pull/2980. We'll have to make a check here to verify that we're not on final leg, and if so skip `legIndex` increment.

Reply:
Thanks for the suggestion! I'll add check here to improve.

Reply:
Assert will occur when we start navigation on both iOS and CarPlay, but do not show waypoint arrival on iOS. For example:
<img width="1680" alt="Screen Shot 2021-05-20 at 2 54 37 PM" src="https://user-images.githubusercontent.com/1496498/118974385-a0594480-b927-11eb-8bc6-f809b381161b.png">

To reproduce it wait for waypoint arrival on iOS and CarPlay, do not dismiss dialog on CarPlay right away and wait till iOS proceeds to the next leg, on only after that press confirmation button.

Reply:
Thanks for the suggestion, I'll work on this part to improve it.